storm-core/src/jvm/org/apache/storm/daemon/Acker.java

```diff
@@ -100,6 +101,8 @@ public void execute(Tuple input) {
             }
             curr.failed = true;
             pending.put(id, curr);
+        } else if(ACKER_RESET_TIMEOUT_STREAM_ID.equals(streamId)) {
```

Comment:
Yes this is very true, in the clj code the entire processing was wrapped in a null check for curr.